src/contexts/UploadContext.tsx

```diff
@@ -0,0 +1,29 @@
+'use client'
+import { createContext, useContext, useState, PropsWithChildren } from 'react'
+
+interface UploadContextProps {
+  triggerUpload: boolean
+  setTriggerUpload: React.Dispatch<React.SetStateAction<boolean>>
```

Comment:
**nitpick:** It should start uploading instead of just toggling boolean. It's worth to pack all uploading logic inside this provider.